src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -529,4 +532,21 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
             authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
         }
     }
+
+    private void setUserRolesForPostings(Posting posting, Course postingCourse) {
+        User author = userRepository.findOneWithGroupsAndAuthoritiesById(posting.getAuthor().getId()).get();
+
+        if (author.getAuthorities().stream().anyMatch(authority -> authority.equals(Authority.INSTRUCTOR_AUTHORITY) || authority.equals(Authority.ADMIN_AUTHORITY))
```

Comment:
The conditions here and for the `else if` case are very long. Can you move them to variables/methods with a sensible name so that is easily understandable which condition is checked?

Reply:
updated